azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/VirtualMachineScaleSetNetworkInterface.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.network;
+
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.network.implementation.NetworkInterfaceInner;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.Resource;
+import com.microsoft.azure.management.resources.fluentcore.model.Refreshable;
+import com.microsoft.azure.management.resources.fluentcore.model.Wrapper;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Virtual machine scale set network interface.
+ */
+@Fluent
+public interface VirtualMachineScaleSetNetworkInterface extends
```

Comment:
would it be possible to derive this from some (new) shared base interface that could be extracted out of the regular NICs and this?  It just looks wrong to repeat so many members for a model that is essentially the same, yet represented by an entirely different type. It seems a common base relationship would be expected here.